etcd/v3/kv_etcd.go

```diff
@@ -1622,8 +1637,12 @@ func (et *etcdKV) listenPeerUrls(ip string, port string) []string {
 }
 
 func (et *etcdKV) constructURL(ip string, port string) string {
-	ip = strings.TrimPrefix(ip, urlPrefix)
-	return urlPrefix + ip + ":" + port
+	prefix := urlPrefix
+	if et.EtcdCommon.IsTLSEnabled() {
+		prefix = urlSecPrefix
+	}
+	ip = strings.TrimPrefix(ip, prefix)
```

Comment:
When caller is including the prefix in the IP, are they sending the right prefix http vs https? Should we trim both prefixes to be safe?

Reply:
I don't know how, I have picked up some of Zoran's changes for TLS.